zerver/views/__init__.py

```diff
@@ -573,6 +573,11 @@ def get_context_data(self, **kwargs):
         except KeyError:
             pass
 
+        try:
+            context['next'] = self.request.GET['next']
+        except KeyError:
+            context['next'] = '/'
+
```

Comment:
Now the url no longer shows `next` but this code is still there because `two_factor` needs it internally. I have, actually, just used `zulip_login_required` to check login status before `otp_required`.